experiments/ClimaEarth/components/ocean/oceananigans.jl

```diff
@@ -0,0 +1,322 @@
+    remapped_F_turb_ρτxz = CC.Remapping.interpolate(sim.remappers.remapper_fc, F_turb_ρτxz)
+    remapped_F_turb_ρτyz = CC.Remapping.interpolate(sim.remappers.remapper_cf, F_turb_ρτyz)
+
+    # set_from_extrinsic_vectors!((; u = oc_flux_u, v = oc_flux_v),
+    #                             sim.model.grid,
+    #                             remapped_F_turb_ρτxz,
+    #                             remapped_F_turb_ρτyz
+    #                             )
+
+    oc_flux_u = surface_flux(sim.sim.model.velocities.u)
+    oc_flux_v = surface_flux(sim.sim.model.velocities.v)
+
+    view(oc_flux_u, :, :, 1) .= remapped_F_turb_ρτxz
+    view(oc_flux_v, :, :, 1) .= remapped_F_turb_ρτyz
+
+    # TODO: Handle this directly when interpolating
+    OC.fill_halo_regions!(oc_flux_u)
+    OC.fill_halo_regions!(oc_flux_v)
+
+    (; ocean_reference_density, ocean_heat_capacity, ocean_fresh_water_density) = sim.ocean_properties
+    remapped_F_lh = CC.Remapping.interpolate(sim.remappers.remapper_cc, F_lh)
+    remapped_F_sh = CC.Remapping.interpolate(sim.remappers.remapper_cc, F_sh)
+    remapped_F_turb_energy = remapped_F_lh + remapped_F_sh
+
+    # TODO: Note, SW radiation penetrates the surface. Right now, we just put
+    # everything on the surface, but later we will need to account for this.
+    # One way we can do this is using directly ClimaOcean
+
+    oc_flux_T = surface_flux(sim.sim.model.tracers.T)
+    view(oc_flux_T, :, :, 1) .=
+        OC.interior(oc_flux_T, :, :, 1) .+ remapped_F_turb_energy ./ (ocean_reference_density * ocean_heat_capacity)
+    OC.fill_halo_regions!(oc_flux_T)
+
+    # Add the part of the salinity flux that comes from the moisture flux, we also need to
+    # add the component due to precipitation (that was done with the radiative fluxes)
+    remapped_F_turb_moisture = CC.Remapping.interpolate(sim.remappers.remapper_cc, F_turb_moisture)
+    oc_flux_S = surface_flux(sim.sim.model.tracers.S)
+    surface_salinity = OC.interior(sim.sim.model.tracers.S, :, :, 1)
+    moisture_fresh_water_flux = remapped_F_turb_moisture ./ ocean_fresh_water_density
+    view(oc_flux_S, :, :, 1) .= OC.interior(oc_flux_S, :, :, 1) .- surface_salinity .* moisture_fresh_water_flux
+
+    OC.fill_halo_regions!(oc_flux_S)
+    return nothing
+end
+
+# function set_from_extrinsic_vectors!(vectors, grid, u, v)
+#     grid = grid
+#     arch = grid.architecture
+#     # TODO: Move this outside
+#     uᶜᶜᶜ = CenterField(grid)
+#     vᶜᶜᶜ = CenterField(grid)
+#     set!(uᶜᶜᶜ, u)
+#     set!(vᶜᶜᶜ, v)
+
+#     # TODO: Change these kernels to be 2D
+#     launch!(arch, grid, :xyz, _rotate_vectors!, uᶜᶜᶜ, vᶜᶜᶜ, grid)
+#     launch!(arch, grid, :xyz, _interpolate_vectors!,
+#             vectors.u, vectors.v, grid, uᶜᶜᶜ, vᶜᶜᶜ)
+#     return nothing
+# end
+
+function Interfacer.update_field!(sim::OceananigansSimulation, ::Val{:area_fraction}, field)
+    Interfacer.remap!(sim.area_fraction, field)
+end
```

Comment:
`area_fraction` is always on the exchange space since it's only used by the coupler so no need to regrid